app/graphql/mutations/integration_collection_mappings/destroy.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Mutations
+  module IntegrationCollectionMappings
+    class Destroy < BaseMutation
+      include AuthenticableApiUser
+      include RequiredOrganization
+
+      graphql_name 'DestroyIntegrationCollectionMapping'
+      description 'Destroy an integration collection mapping'
+
+      argument :id, ID, required: true
+
+      field :id, ID, null: true
+
+      def resolve(id:)
+        integration_collection_mapping = ::IntegrationCollectionMappings::BaseCollectionMapping
+          .joins(:integration)
+          .where(id:)
+          .where(integration: { organization: current_organization }).first
+
+        return not_found_error(resource: 'integration_collection_mapping') unless integration_collection_mapping
```

Comment:
It looks like this logic is already handled in the destroy service. Any reason for having it here as well?